From 9471f3d82b81584c94cf05d4b4d2b53476ffbd23 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 25 Apr 2016 10:35:37 +0800
Subject: [PATCH 1034/5242] MLK-12705 ARM: imx: add support for i.mx7d TO1.2
 busfreq

commit  5803d93ea48629f8ec0598f078bd8d2d9c2fc8b5 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX7D TO1.2 fix the CKE issue, need to follow TO1.0's
precedure for DRAM frequency scaling.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/ddr3_freq_imx7d.S |    8 ++++----
 arch/arm/mach-imx/lpddr3_freq_imx.S |   13 +++++++++----
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-imx/ddr3_freq_imx7d.S b/arch/arm/mach-imx/ddr3_freq_imx7d.S
index 01ab9b0..843e181 100644
--- a/arch/arm/mach-imx/ddr3_freq_imx7d.S
+++ b/arch/arm/mach-imx/ddr3_freq_imx7d.S
@@ -176,8 +176,8 @@
 
 	ldr	r7, [r9, #ANADIG_DIGPROG]
 	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	20f
+	cmp	r7, #0x11
+	bne	20f
 
 	ldr	r7, =0x0
 	str	r7, [r5, #DDRPHY_CA_WLDSKEW_CON0]
@@ -306,8 +306,8 @@
 
 	ldr	r7, [r9, #ANADIG_DIGPROG]
 	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	22f
+	cmp	r7, #0x11
+	bne	22f
 
 	ldr	r7, =0x1d1d1d1d
 	str	r7, [r5, #DDRPHY_CA_WLDSKEW_CON0]
diff --git a/arch/arm/mach-imx/lpddr3_freq_imx.S b/arch/arm/mach-imx/lpddr3_freq_imx.S
index 648d558..fbf3cb0 100644
--- a/arch/arm/mach-imx/lpddr3_freq_imx.S
+++ b/arch/arm/mach-imx/lpddr3_freq_imx.S
@@ -184,8 +184,8 @@
 
 	ldr	r7, [r9, #ANADIG_DIGPROG]
 	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	11f
+	cmp	r7, #0x11
+	bne	11f
 
 	ldr	r7, =0x0
 	str	r7, [r5, #DDRPHY_CA_WLDSKEW_CON0]
@@ -244,6 +244,11 @@
 	ands	r8, r8, #0x4
 	beq	15f
 
+	ldr	r7, [r9, #ANADIG_DIGPROG]
+	and	r7, r7, #0x11
+	cmp	r7, #0x11
+	bne	14f
+
 	ldr	r7, =0x08080808
 	str	r7, [r5, #DDRPHY_CA_DSKEW_CON0]
 	str	r7, [r5, #DDRPHY_CA_DSKEW_CON1]
@@ -255,8 +260,8 @@
 15:
 	ldr	r7, [r9, #ANADIG_DIGPROG]
 	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	13f
+	cmp	r7, #0x11
+	bne	13f
 
 	ldr	r7, =0x1c1c1c1c
 	str	r7, [r5, #DDRPHY_CA_DSKEW_CON0]
-- 
1.7.9.5

